fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -4848,6 +4860,15 @@ public LocalDate getLastRepaymentDate() {
         return currentTransactionDate;
     }
 
+    public LoanTransaction getLastRepaymentTransaction() {
+        for (final LoanTransaction loanTransaction : this.loanTransactions) {
+            if (loanTransaction.isRepaymentType()) {
```

Comment:
missing condition: AND loanTransaction.isNotReversed()

Reply:
Done!